app/controllers/requests_controller.rb

```diff
@@ -0,0 +1,74 @@
+class RequestsController < ApplicationController
+  include RequestsHelper
```

Comment:
Helpers are automatically included in views. This should give you a clue as to their intended function: To abstract complex view logic. Now, I'm not against putting complex controller logic into helpers also in certain circumstances, but for two simple permission methods I don't think I would have. They can go under the `private` directive.

Reply:
So you are saying I should duplicate the methods in this helper, putting them in the controller as well? What if one of the criteria for updating permissions changes? It will have to be updated in multiple places, versus just having this line at the top of the controller to keep the logic in one place. I know there is not much logic, it just seems less clean.

Reply:
But the method names and helper class name make it sound like they are only for this one controller (`RequestsController`) anyways. 

If you have controller behavior that you want to share among _all_ of your controllers, may as well put it the `ApplicationController`.

Abstraction is fine, but you don't have to do it preemptively. If it turns out you need those exact methods in some (but not all) other controllers later, you can always move them then. Otherwise, breaking things up arbitrarily ends up being a hunting expedition for future programmers.

Reply:
Finally, the place for shared behavior among controllers -- if you _do_ need it -- is `app/controllers/concerns`. But...  https://medium.com/@carlescliment/about-rails-concerns-a6b2f1776d7d

Reply:
I personally favor a model predicate method. As for which one, I think I'd choose `User` because it's more flexible.
`@user.can_view(@request)` seems really clean to me.